internal/common/shared/forage.go

```diff
@@ -18,3 +18,21 @@ type ForageDecision struct {
 
 // ForagingDecisionsDict is a map of clients' foraging decisions
 type ForagingDecisionsDict = map[ClientID]ForageDecision
+
+// ShareForageInfomation is used to represent the forage informations shared by agents
+type ShareForageInfomation struct {
+	DecisionMade     ForageDecision
+	ResourceObtained Resources
+	// ShareTo is used to show which agents/clients to SEND the foraging information to
+	ShareTo []ClientID
+	// SharedFrom is used to show where the information came from
+	SharedFrom ClientID
+}
+
+// MakeForagingDict is a map of clients' -> foraging decisions, their resource obtained, and which
+// clients to send this information to
+type MakeForagingDict = map[ClientID]ShareForageInfomation
+
+// ReceiveForagingDict is a map of clients -> array of information of other clients
+// foraging decisions, their resource obtained, and which clients to sent this information
+type ReceiveForagingDict = map[ClientID][]ShareForageInfomation
```

Comment:
Can we name this with `Foragxxx`?

Reply:
Actually, should we change the naming for foraging stuff to follow this one? I think that's a stretch, but at least the types here are `Foragxxx`

Reply:
> Actually, should we change the naming for _all_ foraging stuff to follow this one? I think that's a stretch, but at least the types here are `Foragxxx`

Reply:
Not entirely too sure what to change MakeForagingDict or ReceiveForagingDict to such that its still readable.

Reply:
`ForagingOfferDict` and `ForagingReceiptDict`?

Reply:
Done